app/modules/enduser_setup.c

```diff
@@ -490,8 +520,15 @@ static void enduser_setup_http_urldecode(char *dst, const char *src, int src_len
       }
       *dst++ = c;
     }
+    if ((dst - dst_start) >= dst_len - 1)
```

Comment:
I think you misread the code. It is actually running for escaped and non-escaped sequences.
The only case it's not running for is the last char of a sequence, which is fine, since we'll be exiting the function in that case anyway.

I could be misunderstanding something though.